apps/ewallet/lib/ewallet/release_tasks/config.ex

```diff
@@ -60,17 +61,62 @@ defmodule EWallet.ReleaseTasks.Config do
     Enum.each(@start_apps, &Application.ensure_all_started/1)
     Enum.each(@apps, &ensure_app_started/1)
 
-    case Config.update(%{key => value, originator: %System{}}) do
-      {:ok, [{key, {:ok, _}}]} ->
-        CLI.success("Successfully updated \"#{key}\" to \"#{value}\"")
+    case do_set_config(key, value) do
+      {:ok, setting} ->
+        CLI.success("Updated: `#{key}` is now #{inspect(setting.value)}.")
         :init.stop()
 
-      {:ok, [{key, {:error, :setting_not_found}}]} ->
-        CLI.error("Error: \"#{key}\" is not a valid settings")
+      {:unchanged, value} ->
+        CLI.warn("Skipped: `#{key}` is already set to #{inspect(value)}.")
+        :init.stop()
+
+      {:error, :setting_not_found} ->
+        CLI.error(
+          "Error: `#{key}` is not a valid settings." <>
+            " Please check that the given settings name is correct and settings have been seeded."
+        )
+
+        :init.stop(1)
+
+      {:error, changeset} ->
+        error_message =
+          Enum.reduce(changeset.errors, "", fn {field, {message, _}}, acc ->
+            acc <> "`#{field}` #{message}. "
+          end)
+
+        CLI.error("Error: setting `#{key}` to #{inspect(value)} returned #{error_message}")
         :init.stop(1)
 
       _ ->
         give_up()
     end
   end
+
+  defp do_set_config(key, value) do
+    case Config.get_setting(key) do
+      nil ->
+        {:error, :setting_not_found}
+
+      %{value: existing, type: type} ->
+        case cast_env(value, type) do
+          ^existing -> {:unchanged, existing}
+          casted_value -> do_update(key, value)
```

Comment:
```
warning: variable "casted_value" is unused (if the variable is not meant to be used, prefix it with an underscore)
  lib/ewallet/release_tasks/config.ex:103
```

Reply:
Shouldn't you use the `casted_value` btw?

Reply:
🤦‍♂️